ganttproject/src/main/java/net/sourceforge/ganttproject/storage/ProjectStateStorageFactory.kt

```diff
@@ -0,0 +1,29 @@
+package net.sourceforge.ganttproject.storage
+
+import org.h2.jdbcx.JdbcDataSource
+
+interface ProjectStateStorageFactory {
+  fun getStorage(): ProjectStateStorage
+}
+
+class H2InMemoryStorageFactory : ProjectStateStorageFactory {
+  private companion object {
+    const val H2_IN_MEMORY_URL = "jdbc:h2:mem:gantt-project-state"
```

Comment:
Is it correct that only one project can be opened at any point of time, so that we don't need URL to depend on the project name or another identifier?

Reply:
More important is that projects may be opened and closed one after another, and no matter how many times we do it, there is only one instance of GanttProject class and hence only one instance of your listener

Reply:
> projects may be opened and closed one after another

Because of that I thought it could be better to pass a factory to the listener and use it when a project is opened. However, if only one project is opened at any time, we should be fine with re-using the database, just clearing it on close.

> so generally it is better not to rely on this heavily

If it's needed, we may use a db map instead of a single `ProjectDatabase`. Or add a `project_id` column  to each table. In both cases, the listener needs to be passed the project id on each action.